Additional seeds for TC

aehart opened the pull request:
The TC has been expanded to work for all barrel-only seeds included in the current reference emulation (L1L2, L3L4, and L5L6, but no L2L3 yet).

The other significant change is to the interface: the tracklet projection memories are now grouped into arrays, which are the same length for every TC. The ports corresponding to memories that are not written to by a given TC are successfully removed by the trim_dangling_port option.

Less significantly, I changed the extensions of the files related to the TC processing module to match our style guide. I also added several comments and a few constants to hopefully improve readability a bit.

pwittich commented:
did you git mv the files or delete and add new files? the former is a bit nicer from the continuity point of view...

aehart commented:
I used `git mv`. If you look at the individual commit, you can see the changes represented as moves:
https://github.com/cms-tracklet/firmware-hls/pull/77/commits/243b6b4cc106c7b920a3a96c8fa6aced8f7290fd

pwittich commented:
> I used `git mv`. If you look at the individual commit, you can see the changes represented as moves:
> [243b6b4](https://github.com/cms-tracklet/firmware-hls/commit/243b6b4cc106c7b920a3a96c8fa6aced8f7290fd)

ok thanks, somehow the PR did not notice (probably a limit of the GitHub interface)

pwittich commented on TrackletAlgorithm/TrackletCalculator_calculate_LXLY.h:
```diff
@@ -1,14 +1,15 @@
-#include "TrackletCalculator.hh"
-
 // This function does all the actual calculations for L1L2 TrackletCalculators.
```
should this command change to be LXLY?

pwittich commented:
also can you rebase and squash this PR?

aehart commented:
I fixed the comment and rebased and squashed.

I include three test-benches for TC_L1L2E, TC_L1L2G, and TC_L3L4E. So in principle at least, the CI should run all three. However, only TC_L1L2G has data files in the repository (the files for the others require running download.sh). Although I don't know why running download.sh during the CI would be a problem.

pwittich commented:
@RobertGlein see comment on CI
